eva/parser/lark_visitor/_create_statements.py

```diff
@@ -292,11 +293,24 @@ def create_index(self, tree):
                     table_ref = TableRef(table_name)
                 elif child.data == "index_type":
                     index_type = self.visit(child)
-                elif child.data == "uid_list":
-                    uid_list = self.visit(child)
+                elif child.data == "index_elem":
+                    index_elem = self.visit(child)
+
+        # Parse either a single UDF function call or column list.
+        col_list, udf_func = None, None
+        if not isinstance(index_elem, list):
+            udf_func = index_elem
+
+            # Traverse to the tuple value expression.
+            while not isinstance(index_elem, TupleValueExpression):
```

Comment:
Why do we need this traversal? This should happen automatically in -- https://github.com/georgia-tech-db/eva/blob/f88db49bb3d6f9adbac6a2ae57fc82047a9df467/eva/parser/lark_visitor/__init__.py#L50

Reply:
My goal is to get the column definition for the query. For query that involves UDF expression: e.g., `CREATE INDEX ON table (FeatureExtractor(col));`, the column definition becomes the children tree of `index_elem`, so I need to do a traversal here. Is there a better way to do that in the current parser?

Reply:
I see. You will need to just add a similar function below for `index_elem`: 

```
               elif child.data == "index_type":
                    index_type = self.visit(child)
                elif child.data == "index_elem":
                    index_elem = self.visit(child)

        # Parse either a single UDF function call or column list.
        col_list, udf_func = None, None
        if not isinstance(index_elem, list):
            udf_func = index_elem

            # Traverse to the tuple value expression.
            while not isinstance(index_elem, TupleValueExpression):
                index_elem = index_elem.children[0]
            index_elem = [index_elem]

        col_list = [
            ColumnDefinition(tv_expr.col_name, None, None, None)
            for tv_expr in index_elem
        ]

def index_elem(self, tree):
        uid_list = []
        function_call = None

        for child in tree.children:
            if isinstance(child, Tree):
                if child.data == "uid_list":
                    uid_list = self.visit(child)
                elif child.data == "function_call":
                    function_call = self.visit(child)

        return None
`